presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.agent;
+
+import com.teradata.prestomanager.common.JaxrsParameter;
+import org.apache.logging.log4j.Level;
+
+import javax.ws.rs.core.Response;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.time.chrono.IsoChronology;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.Response.Status;
+
+/**
+ * Class managing retrieval and deletion of Presto logs
+ */
+public class LogsHandler
+{
+    // TODO: Make _all_ of these 'private static final' values configurable
+    private static final Path LOG_DIRECTORY = Paths.get("/tmp", "presto", "logs");
+    private static final Level LOG_ALL = Level.ALL;
+
+    private static final DateTimeFormatter DATE_FORMAT = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive().parseStrict()
+            .append(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+            .appendPattern("Z")
+            .toFormatter().withChronology(IsoChronology.INSTANCE);
+
+    private static final String DATE_GROUP = "date";
+    private static final String LEVEL_GROUP = "level";
+    private static final Pattern LOG_REGEX = Pattern.compile(
+            "^(?<" + DATE_GROUP + ">[0-9]{4}-[0-9]{2}-[0-9]{2}T" +
+                    "[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{3})?[+-][0-9]{4})" +
```

Comment:
One slash to escape the backslash for Java, leaving one behind for the regex. So `\\.` becomes `\.` in the regex.